lib/authentication/auth_mfatoken.js

```diff
@@ -1,41 +0,0 @@
-/*
- * Copyright (c) 2015-2023 Snowflake Computing Inc. All rights reserved.
- */
-
-/**
- * Creates a MFA token authenticator
- *
- * @param {String} token
- *
- * @returns {Object}
- * @constructor
- */
-function AuthMFAToken(connectionConfig) {
-
-  this.password = connectionConfig.password;
-  this.mfaToken = connectionConfig.mfaToken;
-  
-  /**
-     * Update JSON body with token.
-     *
-     * @param {JSON} body
-     *
-     * @returns {null}
-     */
-  this.updateBody = function (body) {
-    body['data']['AUTHENTICATOR'] = 'USERNAME_PASSWORD_MFA';
```

Comment:
The mfaToken authentication file was removed and there is now code which set authenticator as USERNAME_PASSWORD_MFA. Whether this is a deliberate action?

Reply:
The reason why I removed this file is that this authenticator is the same as the default authenticator. On the document, it mentions that "set the authenticator as 'USERNAME_PASSWORD_MFA', so I made the connector use the default authenticator when the user sets the authenticator to 'USERNAME_PASSWORD_MFA'. This implementation is the same as ODBC and JDBC, and I found this when I implemented the libsnowflakeclient passcode authenticator.

Reply:
In my opinionthere is a lack of setting this authenticator type directly. Firstly I see that requested generated for authentication with passcode still contains authenticator = 'SNOWFLAKE':

Reply:
Thank you for your feedback @sfc-gh-pmotacki  @sfc-gh-pbulawa. I refactored the authentication. Please have a look.